staging/src/k8s.io/client-go/tools/cache/controller.go

```diff
@@ -131,11 +132,12 @@ func (c *controller) Run(stopCh <-chan struct{}) {
 		<-stopCh
 		c.config.Queue.Close()
 	}()
-	r := NewReflector(
+	r := NewReflectorWithOptions(
 		c.config.ListerWatcher,
 		c.config.ObjectType,
 		c.config.Queue,
-		c.config.FullResyncPeriod,
+		ReflectorResyncPeriod(c.config.FullResyncPeriod),
+		ReflectorTypeDescription(c.config.ObjectDescription),
```

Comment:
if description is unset, does this do the right thing (fall back to deriving the description from the object)?

Reply:
Yes